services/fipe/automakers.js

```diff
@@ -18,43 +18,23 @@ async function listAutomakers({ vehicleType, referenceTable }) {
     }
   );
 
-  return data
-    .map((item) => ({ nome: item.Label, valor: item.Value }))
-    .sort((a, b) => parseInt(a.valor, 10) - parseInt(b.valor, 10));
+  return sortAutomakers(data);
 }
 
-export async function listCarAutomakers(
-  { referenceTable } = { referenceTable: undefined }
-) {
-  const referenceTableCode =
-    referenceTable || (await getLatestReferenceTable());
+const sortAutomakers = (data) => data
+  .map((item) => ({ nome: item.Label, valor: item.Value }))
+  .sort((a, b) => parseInt(a.valor, 10) - parseInt(b.valor, 10));
 
+const getAutomakers = async (vehicleType, referenceTable) => {
+  if (!referenceTable) {
+    referenceTable = await getLatestReferenceTable();
+  }
   return listAutomakers({
-    vehicleType: VEHICLE_TYPE.CAR,
-    referenceTable: referenceTableCode,
+    vehicleType,
+    referenceTable,
   });
-}
-
-export async function listMotorcycleAutomakers(
-  { referenceTable } = { referenceTable: undefined }
-) {
-  const referenceTableCode =
-    referenceTable || (await getLatestReferenceTable());
-
-  return listAutomakers({
-    vehicleType: VEHICLE_TYPE.MOTORCYCLE,
-    referenceTable: referenceTableCode,
-  });
-}
-
-export async function listTruckAutomakers(
-  { referenceTable } = { referenceTable: undefined }
-) {
-  const referenceTableCode =
-    referenceTable || (await getLatestReferenceTable());
+};
 
-  return listAutomakers({
-    vehicleType: VEHICLE_TYPE.TRUCK,
-    referenceTable: referenceTableCode,
-  });
-}
+export const listCarAutomakers = () => getAutomakers(VEHICLE_TYPE.CAR);
+export const listMotorcycleAutomakers = () => getAutomakers(VEHICLE_TYPE.MOTORCYCLE);
+export const listTruckAutomakers = () => getAutomakers(VEHICLE_TYPE.TRUCK);
```

Comment:
I think we need to modify these interfaces to receive the reference table. And I think it's weird that the test didn't fail, maybe we need to modify the test

![image](https://user-images.githubusercontent.com/16273730/214819567-d53253f5-e752-4667-acba-1085a425d0be.png)

https://github.com/BrasilAPI/BrasilAPI/blob/a7e66373cc930793563f4cf0571fc691d7c5cf7d/pages/api/fipe/marcas/v1/index.js#L31

Reply:
Now you need to fix it